BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.kt

```diff
@@ -0,0 +1,410 @@
+        val json = JSONObject(configurationString)
+        assetsUrl = Json.optString(json, ASSETS_URL_KEY, "")
+        clientApiUrl = json.getString(CLIENT_API_URL_KEY)
+
+        // parse json challenges
+        json.optJSONArray(CHALLENGES_KEY)?.let { challengesArray ->
+            for (i in 0 until challengesArray.length()) {
+                challenges.add(challengesArray.optString(i, ""))
+            }
+        }
+
+        environment = json.getString(ENVIRONMENT_KEY)
+        merchantId = json.getString(MERCHANT_ID_KEY)
+        merchantAccountId = Json.optString(json, MERCHANT_ACCOUNT_ID_KEY, null)
+        analyticsConfiguration = AnalyticsConfiguration.fromJson(json.optJSONObject(ANALYTICS_KEY))
+        braintreeApiConfiguration =
+            BraintreeApiConfiguration.fromJson(json.optJSONObject(BRAINTREE_API_KEY))
+        cardConfiguration = CardConfiguration.fromJson(json.optJSONObject(CARD_KEY))
+        isPayPalEnabled = json.optBoolean(PAYPAL_ENABLED_KEY, false)
+        payPalConfiguration = PayPalConfiguration.fromJson(json.optJSONObject(PAYPAL_KEY))
+        googlePayConfiguration = GooglePayConfiguration.fromJson(json.optJSONObject(GOOGLE_PAY_KEY))
+        isThreeDSecureEnabled = json.optBoolean(THREE_D_SECURE_ENABLED_KEY, false)
+        venmoConfiguration = VenmoConfiguration.fromJson(json.optJSONObject(PAY_WITH_VENMO_KEY))
+        kountConfiguration = KountConfiguration.fromJson(json.optJSONObject(KOUNT_KEY))
+        unionPayConfiguration = UnionPayConfiguration.fromJson(json.optJSONObject(UNIONPAY_KEY))
+        visaCheckoutConfiguration =
+            VisaCheckoutConfiguration.fromJson(json.optJSONObject(VISA_CHECKOUT_KEY))
+        graphQLConfiguration = GraphQLConfiguration.fromJson(json.optJSONObject(GRAPHQL_KEY))
+        samsungPayConfiguration =
+            SamsungPayConfiguration.fromJson(json.optJSONObject(SAMSUNG_PAY_KEY))
+        cardinalAuthenticationJwt = Json.optString(json, CARDINAL_AUTHENTICATION_JWT, null)
+    }
+
+    /**
+     * @return `true` if cvv is required for card transactions, `false` otherwise.
+     */
+    open val isCvvChallengePresent = challenges.contains("cvv")
+
+    /**
+     * @return `true` if postal code is required for card transactions, `false` otherwise.
+     */
+    open val isPostalCodeChallengePresent = challenges.contains("postal_code")
+
+    /**
+     * @return `true` if fraud device data collection should occur; `false` otherwise.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val isFraudDataCollectionEnabled = cardConfiguration.isFraudDataCollectionEnabled
+
+    /**
+     * @return `true` if Venmo is enabled for the merchant account; `false` otherwise.
+     */
+    open val isVenmoEnabled = venmoConfiguration.isAccessTokenValid
+
+    /**
+     * @return the Access Token used by the Venmo app to tokenize on behalf of the merchant.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```

Comment:
This [`RestrictTo`](https://developer.android.com/reference/androidx/annotation/RestrictTo.Scope#LIBRARY_GROUP) is pretty neat - it basically lets us have public access level, but restrict to exposing the documentation to within our SDK.

On iOS, our `nodoc` usage isn't great, since it is not respected if a merchant builds the docs on their own through Xcode. Good news though,[ Swift 5.8 will release a similar feature](https://forums.swift.org/t/nodoc-attribute-for-hiding-symbols-from-the-symbol-graph/59227) to `nodoc` in DocC.